core/domain/rights_manager.py

```diff
@@ -506,22 +592,27 @@ def check_can_access_activity(user, activity_rights):
     Returns:
         bool. Whether the given activity can be accessed by the given user.
     """
+    result: bool = False
     if activity_rights is None:
-        return False
+        result = False
```

Comment:
Why was the original formulation of this code not working?

Reply:
The expected return type of this function is `bool`. So, when we define `bool` as return type, mypy assumes that in every case bool is returned but here we are returning bool only for some cases `by doing elif` and for other case we are not returning anything which causes MyPy to throw error.

So, to return bool for every case I did this change.

Reply:
Thanks for explaining. I think you can keep the old code (it maintains less "state") and add a "return False" at the end.

This is a great thing to catch by typing!

Reply:
return is added at last